apps/davi/src/stores/modules/1_5/fetchers/rpc/useProposalCalls/useProposalCalls.ts

```diff
@@ -0,0 +1,203 @@
+const isApprovalCall = (call: Call) => isApprovalData(call?.data);
+const isZeroHash = (data: string) => data === ZERO_HASH;
+
+const encodeActions = async (
+  calls: Call[],
+  contracts: RichContractData[],
+  chainId: number
+) => {
+  const filteredCalls = calls.filter(
+    call => !isZeroHash(call?.data) || !preventEmptyString(call?.value).isZero()
+  );
+
+  const encodedActions = await Promise.all(
+    filteredCalls.map(async (call: Call) => {
+      if (!!call?.approvalCall) {
+        // If current call is an "spending" call will have a inner approvalCall
+        const { decodedCall: decodedApprovalCall } = await decodeCall(
+          call?.approvalCall,
+          contracts,
+          chainId
+        );
+        // Avoid spreading unnecesary approvalCall;
+        const { approvalCall, ...newCall } = call;
+
+        return {
+          ...newCall,
+          approval: {
+            ...decodedApprovalCall,
+            amount: decodedApprovalCall?.args?._value,
+            token: decodedApprovalCall?.to,
+          },
+        };
+      }
+      return call;
+    })
+  );
+
+  return encodedActions;
+};
+
+type IUseProposalCalls = FetcherHooksInterface['useProposalCalls'];
+
+export const useProposalCalls: IUseProposalCalls = (daoId, proposal) => {
+  const { data: metadata } = useProposalMetadata(proposal?.contentHash);
+
+  const { contracts } = useRichContractRegistry();
+  const { chain } = useNetwork();
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const [options, setOptions] = useState<Option[]>([]);
+
+  const toArray = proposal?.to;
+  const dataArray = proposal?.data;
+  const valuesArray = proposal?.value;
+
+  const calls: Call[] = useMemo(() => {
+    const buildCall = (idx: number): Call => ({
+      from: daoId,
+      to: toArray[idx],
+      data: dataArray[idx],
+      value: valuesArray[idx],
+    });
+
+    return dataArray
+      ?.map((_, index) => {
+        const call = buildCall(index);
+        if (isApprovalData(dataArray[index - 1])) {
+          return {
+            ...call,
+            // We assume that if previous call was an approval, then current one is the one that is being approved
+            // So passing nested temporary approvalCall and remove it from the calls array
+            approvalCall: buildCall(index - 1),
+          };
+        }
+
+        // Inyecting empty call to keep consistency on number of calls per option after nesting approval call. This will be skipped later when decoding options.
+        if (isApprovalCall(call)) return EMPTY_CALL;
+        return call;
+      })
+      .filter(Boolean);
+  }, [daoId, toArray, dataArray, valuesArray]);
+
+  useEffect(() => {
+    if (!daoId || !proposal?.id || !calls) {
+      setOptions([]);
+      return () => {};
+    }
+
+    let cancelled = false;
+
+    // TODO: useTotalLocked hook when ready
+    const totalLocked = BigNumber.from(1000000000000000); //! HARDCODED

```

Comment:
useTotalLocked is implemented in my pr https://github.com/DXgovernance/davi-monorepo/pull/227 so we could remove this todo if we merge my pr first